addons/main/functions/UnitAction/fnc_doHide.sqf

```diff
@@ -27,7 +27,8 @@ if (
 ) exitWith {false};
 
 // do nothing when already inside
-if (RND(GVAR(indoorMove)) && {_unit call FUNC(isIndoor)}) exitWith {
+if (RND(GVAR(indoorMove)) && { lineIntersects [ eyePos _unit, ( eyePos _unit ) vectorAdd [0, 0, 10] ] } ) exitWith {
```

Comment:
Knowing even the server part doesn't run extremely well (especially if not a dedicated, and even more especially if on the same machine as the client), saving the `eyePos` probly would make it faster and probly avoid an instance where both statements differ:
```suggestion
if (RND(GVAR(indoorMove)) && { private _eyePos = eyePos _unit; lineIntersects [ _eyePos, _eyePos vectorAdd [0, 0, 10] ] } ) exitWith {
```